app/controllers/concerns/searchable.rb

```diff
@@ -0,0 +1,47 @@
+module Searchable
+  extend ActiveSupport::Concern
+
+  private
+
+  def sort_params
+    params.require(:data).permit(
+      :collection_id, :sort_id, :text,
+      :institution_id,
+      :theme
+    )
+  end
+
+  def select_institution_id
+    # if the institution_id is != 0 then get it
+    # if the collection id is != 0 then get that collections institution id
+    # if the collection id == 0, then make the institution id also as 0
+    if sort_params[:institution_id].to_i > 0
+      sort_params[:institution_id].to_i
+    else
+      Collection.where(id: sort_params[:collection_id]).
+        first&.institution_id.to_i
+    end
+  end
+
+  def load_institutions
+    new_institution = Institution.new(id: 0, name: "All Institutions")
+    collection_id = params[:data] && sort_params.fetch(:collection_id)
+    @institutions = if collection_id.to_i == 0
+                      Institution.all
+                    else
+                      Institution.joins(:collections).
+                        where("collections.id = ?", collection_id)
+                    end.to_a.unshift(new_institution)
+  end
+
+  def collection
```

Comment:
I realise this is existing code moved out of the controller, but seems like `load_collection` would be more consistent naming here, given the method above performs a very similar action